redisq/queue.go

```diff
@@ -22,6 +22,7 @@ const batchSize = 100
 
 type redisStreamClient interface {
 	Del(keys ...string) *redis.IntCmd
+	TxPipeline() redis.Pipeliner
```

Comment:
Shame this mess with the minimal `redisStreamClient` interface. Could haves something like:

```go
Transact(tx func(redisStreamClient) error) error
```

And wrap the Redis client:

```go
type client struct {
  *redis.Client
}

func (c *client) Transact(exe func(pipe redisSteamClient) error) error {
  pipe := c.TxPipeline()
  err := exe(pipe)
  if err != nil {
    return err
  }
  _, err = pipe.Exec()
  return err
}
```